RFC-0043-torch-native-runtime.md

```diff
@@ -0,0 +1,266 @@
+  ModelRunner model_runner(FILENAME, MODEL_NAME, executor_type, runtime_config, placement);
+
+  // args and kwargs should have the same PyTree spec as the example inputs from Python
+  std::vector<c10::IValue> args = ...;
+  std::unordered_map<std::string, c10::IValue> kwargs = ...;
+
+  c10::IValue result = model_runner.run(args, kwargs); // safe to be called from multiple threads
+
+  return 0;
+}
+```
+
+Therefore here is the list of types we expose as TORCH_API:
+- ModelRunner (and its base class ModelRunnerBase) (from torch/csrc/nativert/ModelRunner.h)
+  - The main execution engine and entry point for exported graph.
+- BaseRuntimeConfig (from torch/csrc/nativert/executor/ModelRunnerBase.h)
+  - Centralized type to store configs/knobs.
+- ExecutorType (from torch/csrc/nativert/executor/ModelRunnerBase.h)
+  - The delegate executor type to be used for execution. Currently supports: 1. Plain interpreter (default) 2. AOTInductor 3. MTIA delagate.
+- Placement (from torch/csrc/nativert/executor/Placement.h)
+  - Typically not used. Sometimes useful for adjusting the device of weights during model loading.
+
+### Code layout
+
+There are mainly two parts of sigmoid: C++ part and Python part.
+- Currently we put all the C++ code under `torch/nativert`. All the function/classes are put under c++ namespace `torch::nativert`
+- Python code are put under `torch/export/`. Python code is needed for packaging torch.export() artifacts into *PT2 Archive* and later loaded by the C++ runtime.
+
+## **Metrics**
+We think it will be a good start to accomplish the following in 2025:
+1. Feature parity: nativert can work on a set of jit-tracable models. (H1)
+2. OSS Coverage: We can run nativert on a list of OSS models. (H1)
+   a. Criteria: Anything working on HuggingFace Optimum should be a good candidate.
+3. Community: We have 1 oss customer adopting nativert in their e2e workflow. (H2)
+
+## **FAQ**
+
+### “Are we targeting Desktop/PC use cases here?”
+
+* We will target "commodity hardware", meaning the architecture is vendored by >= 2 vendors. Example: x64, CUDA/ROCM.
+* We intend to target common PC environments via ExecuTorch being the frontend API surface.
+
+### “What is your overall strategy for deprecating TorchScript in OSS?”
+
+We plan to:
+
+* Make a clear announcement to oss users that TorchScript is deprecated, in favor of nativert. But we will not force deprecate Torchscript but focus on providing a better solution.
+* Leverage major partners, ONNX, TRT, Executorch, HF, who are actively moving users away from Torchscript with their respective unique competitive advantage as incentive for users to migrate.
+
+### “Does your deprecation strategy mean that we will pretty much never be able to delete TorchScript?”
+
+* Possibly. But it is more likely that once nativert is prominent enough, we could delete it. Those users who stay with TorchScript will need to stay with an older PyTorch version.
+
```

Comment:
Is this needed?

Reply:
My reasoning is that some people might be interested in replacing TorchScript with nativert, therefore we should briefly mention our plan with TorchScript here as well, since nativert provides additional value by succeeding TorchScript in oss.